src/move_embargo_args.py

```diff
@@ -78,80 +105,240 @@ def parse_args():
     registry = butler.registry
     dest_butler = Butler(namespace.torepo, writeable=True)
     dest_registry = dest_butler.registry
-    datasetTypeList = namespace.datasettype
 
     # Initialize the logger and set the level
     CliLog.initLog(longlog=True)
     CliLog.setLogLevels(logLevels=[(None, namespace.log)])
     # CliLogNew.initLog(log=namespace.log)
     logger = logging.getLogger("lsst.transfer.embargo")
     logger.info("log level %s", namespace.log)
-    logger.info("whats the datasettypelist in here: %s", datasetTypeList)
-    collections = namespace.collections
+    logger.info("namespace: %s", namespace)
+
+    # If move is true, then you'll need write
+    # permissions from the fromrepo (embargo)
+    # For now we will not allow move to be true
     if namespace.move:
         raise ValueError(
             "namespace.move is True. Program terminating because this is too dangerous."
         )
 
+    # determine if we will use the config file or the
+    # provided datasettypelist and collections args
+    # from the cli
+    if namespace.use_dataquery_config:
+        # define the config file
+        config_file = namespace.dataquery_config_file
+        logger.info(
+            "using the config file, config_file name/path is %s", config_file)
+        # read config file
+        with open(config_file, "r") as f:
+            config = yaml.safe_load(f)
+        logger.info("config %s", config)
+        # Extract datasettype and collections from config
+        datasetTypeList = []
+        collections = []
+        embargohrs_list = []
+        for query in config["dataqueries"]:
+            datasetTypeList.append(query["datasettype"])
+            collections.append(query["collections"])
+            if "embargohrs" in query:
+                embargohrs_list.append(query["embargohrs"])
+    else:
+        datasetTypeList = namespace.datasettype
+        collections = namespace.collections
+
+    logger.info("whats the datasettypelist in here: %s", datasetTypeList)
     move = namespace.move
     dest_uri_prefix = namespace.desturiprefix
     # Dataset to move
     dataId = {"instrument": namespace.instrument}
-    # Define embargo period
-    embargo_period = astropy.time.TimeDelta(
-        namespace.embargohours * 3600.0, format="sec"
-    )
-    if namespace.nowtime != "now":
-        now = astropy.time.Time(namespace.nowtime, scale="tai", format="iso")
+    # Define embargo period and nowtime
+    logger.info("embargo  hrs: %s", namespace.embargohours)
+    logger.info("past embargo hrs: %s", namespace.pastembargohours)
+    logger.info("nowtime: %s", namespace.nowtime)
+
+    # option for embargohours and nowtime to be individual items
+    # or lists
+    # we would like for these arguments to be individual items
+    # if passed via cli, but they may be lists if passed via config yaml
+    if len(namespace.embargohours) == 1 and len(namespace.nowtime) == 1:
```

Comment:
I think the simplest way to combine this code with the next clause is to convert each argument to a single-element list.

Reply:
Did so.